SQF/dayz_server/compile/server_updateObject.sqf

```diff
@@ -18,173 +20,163 @@ _firstTime = false;
-			};
-			//diag_log ("HIVE: WRITE: "+ str(_key));
-			_key call server_hiveWrite;
-		};
+	_inventory = [
+		getWeaponCargo _object,
+		getMagazineCargo _object,
+		getBackpackCargo _object
+	];
+	_previous = str (_object getVariable ["lastInventory", []]);
+	if (str _inventory != _previous) then {
+		_object setVariable ["lastInventory", _inventory];
+		_key = if (_objectID == "0") then { format ["CHILD:309:%1:", _uid] + str _inventory } else { format ["CHILD:303:%1:", _objectID] + str _inventory };
+		_key call server_hiveWrite;
+	};
 };
 
 _object_damage = {
-	private["_hitpoints","_array","_hit","_selection","_key","_damage"];
-		_hitpoints = _object call vehicle_getHitpoints;
-		_damage = damage _object;
-		_array = [];
-		{
-			_hit = [_object,_x] call object_getHit;
-			_selection = getText (configFile >> "CfgVehicles" >> (typeOf _object) >> "HitPoints" >> _x >> "name");
-			if (_hit > 0) then {_array set [count _array,[_selection,_hit]]};
-			_object setHit [_selection, _hit];
-		} count _hitpoints;
-	
-		_key = format["CHILD:306:%1:%2:%3:",_objectID,_array,_damage];
-		//diag_log ("HIVE: WRITE: "+ str(_key));
-		_key call server_hiveWrite;
-	_object setVariable ["needUpdate",false,true];
-	};
+	private ["_hitpoints","_array","_hit","_selection","_key","_damage"];
+	_hitpoints = _object call vehicle_getHitpoints;
+	_damage = damage _object;
+
+	_array = [];
+	{
+		_hit = [_object, _x] call object_getHit;
+		_selection = getText (configFile >> "CfgVehicles" >> typeOf _object >> "HitPoints" >> _x >> "name");
+		if (_hit > 0) then {
+			_array set [count _array, [_selection,_hit]];
+		};
+		_object setHit [_selection, _hit];
+	} count _hitpoints;
+
+	_key = "CHILD:306:" + _objectID + ":" + str _array + ":" + str _damage;
+	_key call server_hiveWrite;
+	_object setVariable ["needUpdate", false, true];
+};
 
 _object_killed = {
-	private["_hitpoints","_array","_hit","_PUID","_selection","_key","_damage"];
+	private ["_hitpoints","_array","_hit","_PUID","_selection","_key","_damage"];
 	_hitpoints = _object call vehicle_getHitpoints;
-	//_damage = damage _object;
 	_damage = 1;
+
 	_array = [];
 	{
-		_hit = [_object,_x] call object_getHit;
-		_selection = getText (configFile >> "CfgVehicles" >> (typeOf _object) >> "HitPoints" >> _x >> "name");
-		if (_hit > 0) then {_array set [count _array,[_selection,_hit]]};
-		_hit = 1;
-		_object setHit [_selection, _hit];
+		_hit = [_object, _x] call object_getHit;
+		_selection = getText (configFile >> "CfgVehicles" >> typeOf _object >> "HitPoints" >> _x >> "name");
+		if (_hit > 0) then {
+			_array set [count _array, [_selection, _hit]];
+		};
+		_object setHit [_selection, 1];
 	} count _hitpoints;
-	
-	if (_objectID == "0") then {
-		_key = format["CHILD:306:%1:%2:%3:",_uid,_array,_damage];
-	} else {
-		_key = format["CHILD:306:%1:%2:%3:",_objectID,_array,_damage];
-	};
-	//diag_log ("HIVE: WRITE: "+ str(_key));
+
+	_key = "CHILD:306:" + _objectID + ":" + str _array + ":" + str _damage;
```

Comment:
If you look it's using the same 306 call for the objectID and objectUID which isn't correct. Vehicles also only use the objectID, they don't use objectUID.